backend/openapi.yaml

```diff
@@ -1032,20 +1031,875 @@ paths:
+        '400':
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error400Response'
+        '401':
+          description: Unauthenticated
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error401Response'
+        '404':
+          description: Not Found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error404Response'
+        '500':
+          description: Internal Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error500Response'
+
+  /campaign/{campaignId}/govsg:
+    get:
+      security:
+        - bearerAuth: []
+      tags:
+        - GovSG
+      summary: Get GovSG campaign details
+      parameters:
+        - name: campaignId
+          in: path
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: GovSG campaign details
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GovSGCampaign'
+        '401':
+          description: Unauthenticated
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error401Response'
+        '403':
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error403Response'
+        '500':
+          description: Internal Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error500Response'
+  /campaign/govsg/templates:
+    get:
+      security:
+        - bearerAuth: []
+      tags:
+        - GovSG
+      summary: Get available templates for GovSG channel
+      responses:
+        200:
+          description: Success
+          content:
+            application/json:
+              schema:
+                required:
+                  - data
+                properties:
+                  data:
+                    $ref: '#/components/schemas/GovSGTemplate'
+        '401':
+          description: Unauthenticated
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error401Response'
+        '403':
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error403Response'
+        '500':
+          description: Internal Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error500Response'
+  /campaign/{campaignId}/govsg/upload/start:
```

Comment:
Qn: Sorry if it sounds a bit RTFM but, how are `campaignId`s generated?

When I was trying the product out, the IDs look like incrementing numbers to me.

And as such, I don't know why we need different paths for different `campaignId` seeing that we could grab all the metadata we need with `campaignId` alone. Is it a legacy thing?

Reply:
There're a bunch of `Campaigns` endpoints that are for creating, updating campaigns etc. The IDs are generated from there.

Reply:
> how are campaignIds generated

yeah it's auto-incremented primary key in db

> I don't know why we need different paths for different campaignId seeing that we could grab all the metadata we need with campaignId alone

not sure I fully understand this question, but there are a bunch of paths for campaigns that are triggered at different points of creating the campaign (i.e. is not just about grabbing metadata).

Reply:
@jia1 , we can do that, no reason not too. Here I'm just sticking with the current convention for consistency. We can refactor this later, but the impact is rather minimal rn that it could be overlooked (namely just a few extra lines in the route files)